benchmarks/scripts.py

```diff
@@ -55,21 +55,32 @@ def circuit_benchmark(nqubits, backend, circuit_name, circuit_options=None,
         result = result.numpy()
     logs.log(dry_run_transfer_time=time.time() - start_time)
     dtype = str(result.state().dtype)
-    del(result)
+    del result
+    del circuit
 
-    simulation_times, transfer_times = [], []
+    creation_times, simulation_times, transfer_times = [], [], []
     for _ in range(nreps):
+        start_time = time.time()
+        circuit = qibo.models.Circuit(nqubits)
+        circuit.add(gates)
+        if nshots is not None:
+            # add measurement gates
+            circuit.add(qibo.gates.M(*range(nqubits)))
+        creation_times.append(time.time() - start_time)
         start_time = time.time()
         result = circuit(nshots=nshots)
         simulation_times.append(time.time() - start_time)
         start_time = time.time()
         if transfer:
             result = result.numpy()
         transfer_times.append(time.time() - start_time)
-        del(result)
+        del result
+        del circuit

```

Comment:
Thank you @migueldiascosta for running these benchmarks on different hardware.

Even though this is already merged, just to provide some additional information

> these changes work around the issue for me by always destroying the circuit, which frees the GPU memory, and always recreating it (but I suppose an alternative would be to avoid recreating the circuits by e.g. adding a method to the circuit class that clears the state, or making its constructor overwrite any existing state?)

Indeed, I believe the issue here is that the circuit is [caching the final state of the last execution](https://github.com/qiboteam/qibo/blob/df5b00ba0dd1042c97687e6b9a150ef478bd2164/src/qibo/models/circuit.py#L1024). If that is the case, a potential work around that would avoid deleting the whole circuit would be
```py
del circuit._final_state
```
(or maybe `circuit._final_state = None`), but I have not tested if this indeed works. In any case, this is manipulating a private attribute so it is probably better if we do it internally in the `Circuit` object and provide a method that clears the state, as you propose.